src/app/modules/auth/auth.middlewares.ts

```diff
@@ -53,3 +57,32 @@ export const denyRpSpStudentEmails: ControllerHandler<
         .json({ message: 'User not found' }),
     )
 }
+
+/**
+ * Logs all admin actions which change database state (i.e. non-GET requests)
+ * @returns next
+ */
+export const logAdminAction: ControllerHandler<
+  unknown,
+  unknown,
+  unknown
+> = async (req, res, next) => {
```

Comment:
actually we don't need the `unknown` types since our custom `ControllerHandler` defaults these args to `unknown`

Reply:
thanks! removed them; also for denyRpSpStudentEmails :)